pkg/pilot/cassandra/v3/pilot.go

```diff
@@ -80,9 +82,39 @@ func (p *Pilot) CmdFunc(pilot *v1alpha1.Pilot) (*exec.Cmd, error) {
 }
 
 func (p *Pilot) syncFunc(pilot *v1alpha1.Pilot) error {
-	if p.genericPilot.IsThisPilot(pilot) {
+	if !p.genericPilot.IsThisPilot(pilot) {
+		return nil
 	}
-	return nil
+	pilot, err := p.pilotLister.Pilots(pilot.Namespace).Get(pilot.Name)
+	if err != nil {
+		return err
+	}
+	pilotCopy := pilot.DeepCopy()
+	if pilotCopy.Status.Cassandra == nil {
+		pilotCopy.Status.Cassandra = &v1alpha1.CassandraPilotStatus{}
+	}
+
+	var errs []error
+
+	version, err := p.nodeTool.Version()
+	if err == nil {
+		lastReportedVersion := pilotCopy.Status.Cassandra.Version
+		if lastReportedVersion == nil || !lastReportedVersion.Equal(*version) {
+			pilotCopy.Status.Cassandra.Version = version
+		}
+	} else {
+		errs = append(errs, err)
+		pilotCopy.Status.Cassandra.Version = nil
+	}
+	if !reflect.DeepEqual(pilot, pilotCopy) {
```

Comment:
We should avoid DeepEqual as much as possible - it's slow and expensive. A simple bool for needsUpdate would suffice, especially as this block is quite simple and small at this point.